### PR TITLE
feat(sdk): add new approveAndExecuteSwap method

### PR DESCRIPTION
**Pull request by GabrielBuragev**

### CODE REVIEW

File: packages/sdk/src/swap/sdk.ts [hunk 1]
```diff
@@ -285,20 +285,33 @@ export class SwapSDK {
...
 
+    if (!isGasAsset(internalAsset)) {
+      const allowanceInfo = await checkVaultAllowance(params, {
+        signer,
+        network,
+      });
+
+      if (!allowanceInfo.hasSufficientAllowance) {
+        await this.approveVault(params, txOpts);
+      }
```
Comment (acdibble) [inline packages/sdk/src/swap/sdk.ts:305]: approve vault already checks for sufficient allowance
```suggestion
      await this.approveVault(params, txOpts);
```

File: packages/sdk/src/swap/sdk.ts [hunk 2]
```diff
@@ -350,6 +350,32 @@ export class SwapSDK {
...
+
+    let approveTransaction = null;
+    if (!isGasAsset(internalAsset)) {
+      approveTransaction = await this.approveVault(params, txOpts);
+    }
+
+    const swapTransaction = await this.executeSwap(params, txOpts);
+
+    return {
+      approveTransaction,
```
Comment (niklasnatter) [inline packages/sdk/src/swap/sdk.ts:374]: would call this `approveTxRef` to be consistent with the new status naming

File: packages/sdk/src/swap/sdk.ts [hunk 3]
```diff
@@ -350,6 +350,32 @@ export class SwapSDK {
...
+      swapTransaction,
```
Comment (niklasnatter) [inline packages/sdk/src/swap/sdk.ts:375]:
```suggestion
      swapTxRef,
```

File: packages/sdk/src/swap/sdk.ts [hunk 4]
```diff
@@ -350,6 +350,32 @@ export class SwapSDK {
     if (!result.success) throw new Error(result.reason);
   }
 
+  async approveAndExecuteSwap(
+    params: ExecuteSwapParams,
+    txOpts: TransactionOptions & { signer?: Signer } = {},
```
Comment (acdibble) [inline packages/sdk/src/swap/sdk.ts:355]: should we accept nonces or maybe a function to get the next nonce?

**Review by acdibble — APPROVED** [review]

File: packages/sdk/src/swap/sdk.ts (hunk 4 above)
Comment (GabrielBuragev) [reply packages/sdk/src/swap/sdk.ts]: As agreed, ignoring nonces for now.

**Review by niklasnatter — APPROVED** [review]

**Pull request merged**